### PR TITLE
🗼Revise "softmax-regression-scratch_vn" - Phần 2

### LINKED ISSUE #920: Revise "softmax-regression-scratch_vn" - Phần 2
Phần này được dịch bởi:
* **lamngoctan**
* **sudohainguyen**

Nếu bạn đã dịch phần này, vui lòng bỏ qua việc revise.

### PR DESCRIPTION
**Pull request by duythanhvn**
Close #920

### CODE REVIEW

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 1]
```diff
@@ -190,7 +190,7 @@ Note that we flatten each original image in the batch into a vector with length
 
 Bây giờ chúng ta đã định nghĩa hàm softmax, chúng ta có thể bắt đầu lập trình mô hình hồi quy softmax.
 Đoạn mã sau định nghĩa lượt truyền xuôi thông qua mạng.
-Chú ý rằng chúng ta làm phẳng mỗi ảnh gốc trên tập lưu trữ bằng một vector có độ dài `num_inputs` bằng hàm `reshape` trước khi truyền dữ liệu sang mô hình đã khởi tạo.
+Chú ý rằng chúng ta sẽ làm phẳng mỗi ảnh gốc trên batch bằng một vector có độ dài `num_inputs` bằng hàm `reshape` trước khi truyền dữ liệu sang mô hình đã khởi tạo.
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Chú ý rằng chúng ta sẽ làm phẳng mỗi ảnh gốc trên batch thành một vector có độ dài `num_inputs` bằng hàm `reshape`, trước khi truyền dữ liệu sang mô hình đã khởi tạo.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 2]
```diff
@@ -283,11 +283,10 @@ Taking the mean yields the desired result.
 -->
 
 Độ chính xác được tính toán như sau:
-Đầu tiên, lệnh `y_hat.argmax(axis=1)` được thực thi nhằm lấy ra các lớp được dự đoán (cho bởi chỉ số của các phần tử lớn nhất của mỗi hàng).
-Kết quả trả về sẽ có cùng kích thước với biến `y`
-và bây giờ ta chỉ cần so sánh hai vector này.
-Vì toán tử `==` so khớp cả kiểu dữ liệu của biến (ví dụ một biến `int` và một biến `float32` không thể bằng nhau), ta cần phải ép kiểu chúng một cách thống nhất (ở đây ta chọn kiểu `float32`).
-Kết quả sẽ là một `ndarray` chứa các giá trị 0 (false) và 1 (true).
+Đầu tiên, lệnh `y_hat.argmax(axis=1)` được thực thi nhằm lấy ra các lớp được dự đoán (được cho bởi chỉ số của các phần tử lớn nhất của mỗi hàng).
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Đầu tiên, lệnh `y_hat.argmax(axis=1)` được thực thi nhằm lấy ra các lớp được dự đoán (được cho bởi chỉ số của phần tử lớn nhất trên mỗi hàng).
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 3]
```diff
@@ -218,10 +218,10 @@ we can use the `pick` function which allows us to easily select the appropriate
 Below, we illustrate the `pick` function on a toy example, with $3$ categories and $2$ examples.
 -->
 
-Nhắc lại rằng entropy chéo lấy kết quả là hàm đối log hợp lý của xác suất dự đoán được gán cho nhãn thực $-\log P(y \mid x)$.
+Nhắc lại rằng entropy chéo lấy kết quả là hàm đối log hợp lý của xác suất dự đoán được gán cho nhãn thật $-\log P(y \mid x)$.
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Nhắc lại rằng hàm entropy chéo lấy đầu vào là đối log hợp lý của xác suất dự đoán được gán cho nhãn thật $-\log P(y \mid x)$.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 4]
```diff
@@ -218,10 +218,10 @@ we can use the `pick` function which allows us to easily select the appropriate
...
 Thay vì lặp qua các dự đoán của mô hình bằng vòng lặp `for` trong Python (có xu hướng kém hiệu quả),
-chúng ta có thể sử dụng hàm `pick` mà cho phép ta chọn lựa dễ dàng các phần tử thích hợp từ ma trận của các biến softmax đầu vào.
-Dưới đây, hàm `pick` được sử dụng như một ví dụ đơn giản với ma trận $2$ hàng $3$ cột.
+chúng ta có thể sử dụng hàm `pick` mà cho phép ta dễ dàng chọn các phần tử thích hợp từ ma trận của các biến softmax ban đầu.
+Dưới đây, hàm `pick` sẽ được sử dụng như một ví dụ đơn giản với ma trận có $3$ lớp và $2$ mẫu.
```
Comment (lkhphuc) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Dưới đây, ta minh hoạ cách sử dụng hàm `pick` trong một ví dụ đơn giản với ma trận có $3$ lớp và $2$ mẫu.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 5]
```diff
@@ -209,7 +209,7 @@ This may be the most common loss function in all of deep learning because, at th
 -->
 
 Tiếp đến chúng ta cần lập trình hàm mất mát entropy chéo đã được giới thiệu ở :numref:`sec_softmax`.
-Đây có lẽ là hàm mất mát thông dụng nhất trong nghiên cứu về học sâu vì hiện nay số lượng bài toán phân loại vượt trội hơn số lượng bài toán hồi quy.
+Đây có lẽ là hàm mất mát thông dụng nhất trong tất cả nghiên cứu về học sâu vì hiện nay số lượng bài toán phân loại đã vượt xa hơn số lượng bài toán hồi quy.
```
Comment (ngcthuong) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Đây có lẽ là hàm mất mát thông dụng nhất trong phần lớn nghiên cứu về học sâu vì hiện nay số lượng bài toán phân loại đã vượt xa hơn số lượng bài toán hồi quy.
```
Câu này dùng tất cả là sai, cross-entropy chỉ dùng trong bài toáng nhận dạng thôi. Còn rất nhiều bài toán khác nữa.

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 6]
```diff
@@ -218,10 +218,10 @@ we can use the `pick` function which allows us to easily select the appropriate
 Below, we illustrate the `pick` function on a toy example, with $3$ categories and $2$ examples.
 -->
 
-Nhắc lại rằng entropy chéo lấy kết quả là hàm đối log hợp lý của xác suất dự đoán được gán cho nhãn thực $-\log P(y \mid x)$.
+Nhắc lại rằng entropy chéo lấy kết quả là hàm đối log hợp lý của xác suất dự đoán được gán cho nhãn thật $-\log P(y \mid x)$.
 Thay vì lặp qua các dự đoán của mô hình bằng vòng lặp `for` trong Python (có xu hướng kém hiệu quả),
-chúng ta có thể sử dụng hàm `pick` mà cho phép ta chọn lựa dễ dàng các phần tử thích hợp từ ma trận của các biến softmax đầu vào.
-Dưới đây, hàm `pick` được sử dụng như một ví dụ đơn giản với ma trận $2$ hàng $3$ cột.
+chúng ta có thể sử dụng hàm `pick` mà cho phép ta dễ dàng chọn các phần tử thích hợp từ ma trận của các biến softmax ban đầu.
```
Comment (ngcthuong) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
chúng ta có thể sử dụng hàm `pick` để dễ dàng chọn các phần tử thích hợp từ ma trận của các giá trị softmax.
```
Đoạn này entries cũng có nghĩa là các thành phần phần từ của ma trận.

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 7]
```diff
@@ -283,11 +283,10 @@ Taking the mean yields the desired result.
 -->
 
 Độ chính xác được tính toán như sau:
-Đầu tiên, lệnh `y_hat.argmax(axis=1)` được thực thi nhằm lấy ra các lớp được dự đoán (cho bởi chỉ số của các phần tử lớn nhất của mỗi hàng).
-Kết quả trả về sẽ có cùng kích thước với biến `y`
-và bây giờ ta chỉ cần so sánh hai vector này.
-Vì toán tử `==` so khớp cả kiểu dữ liệu của biến (ví dụ một biến `int` và một biến `float32` không thể bằng nhau), ta cần phải ép kiểu chúng một cách thống nhất (ở đây ta chọn kiểu `float32`).
-Kết quả sẽ là một `ndarray` chứa các giá trị 0 (false) và 1 (true).
+Đầu tiên, lệnh `y_hat.argmax(axis=1)` được thực thi nhằm lấy ra các lớp được dự đoán (được cho bởi chỉ số của phần tử lớn nhất trên mỗi hàng).
```
Comment (ngcthuong) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Đầu tiên, dùng lệnh `y_hat.argmax(axis=1)` nhằm lấy ra các lớp được dự đoán (được cho bởi chỉ số của phần tử lớn nhất trên mỗi hàng).
```
